ateam_ai/test/trajectory_generation/trapezoidal_motion_profile_test.cpp

```diff
@@ -166,3 +166,111 @@ TEST(TrapezoidalMotionProfile, Trajectory1d_ShouldReturnValidTrajectory_WhenShor
 
   ASSERT_TRUE(IsValidTrajectory(ret, max_vel, max_accel, dt));
 }
+
+TEST(TrapezoidalMotionProfile, Trajectory3d_ShouldReturnOneLong_WhenAlreadyAtTarget)
+{
+  Eigen::Vector3d start{10, 10, 1};
+  Eigen::Vector3d end{10, 10, 1};
+  Eigen::Vector3d start_vel{10, 10, 1};
+  Eigen::Vector3d end_vel{10, 10, 1};
+  Eigen::Vector3d max_vel{10, 10, 1};
+  Eigen::Vector3d max_accel{10, 10, 1};
+  double dt = 0.1;
+  double current_time = 1;
+
+  Trajectory ret = TrapezoidalMotionProfile::Generate3d(
+    start,
+    start_vel,
+    end,
+    end_vel,
+    max_vel,
+    max_accel,
+    dt,
+    current_time);
+
+  ASSERT_GT(ret.samples.size(), 0);
+  EXPECT_EQ(ret.samples.front().pose, start);
+  EXPECT_EQ(ret.samples.front().pose, end);
+  EXPECT_EQ(ret.samples.front().vel, start_vel);
+  EXPECT_EQ(ret.samples.front().vel, end_vel);
+  Eigen::Vector3d zero{0, 0, 0};  // Get around #define weirdness
+  EXPECT_EQ(ret.samples.front().accel, zero);
```

Comment:
Did [`Eigen::Vector3d::Zero()`](https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#a422ddeef58bedc7bddb1d4357688d761) not work here?

```suggestion
  EXPECT_EQ(ret.samples.front().accel, Eigen::Vector3d::Zero());
```

Reply:
Yeah, that seems to work. Something is up with the brace initialized vector adding 2 extra args, probably something to do with the commas and #define not playing nice